tools/BuildPackagesTask/Microsoft.Azure.Build.Tasks.Test/TestSetGeneratorTester.cs

```diff
@@ -0,0 +1,513 @@
+                "src/ResourceManager/StreamAnalytics/",
+                "src/path2",
+                "random3"
+            };
+            string mapFilePath = GetMapFilePath();
+            HashSet<string> expected = new HashSet<string>()
+            {
+                "test1.dll",
+                "test2.dll",
+                "test3.dll"
+            };
+            int expectedNumberFiles = 51;
+            HashSet<string> actual;
+
+            //act
+            actual = (HashSet<string>)(TestSetGenerator.GetTests(paths, mapFilePath));
+
+            //assert            
+            Assert.True(expectedNumberFiles == actual.Count);
+        }
+
+        [Fact]
+        [Trait(AcceptanceType, CheckIn)]
+        public void GetTests_MultiplePathsAndMultipleMappingsWithMatchingPaths_ReturnsMatchingTests()
+        {
+            //arrange
+            HashSet<string> paths = new HashSet<string>()
+            {
+                "src/ResourceManager/StreamAnalytics/file",
+                "src/ResourceManager/Websites/"
+            };
+            string mapFilePath = GetMapFilePath();
+            int expectedNumberFiles = 2;
+            HashSet<string> actual;
+
+            //act
+            actual = (HashSet<string>)(TestSetGenerator.GetTests(paths, mapFilePath));
+
+            //assert            
+            Assert.True(expectedNumberFiles == actual.Count);
+        }
+
+        [Fact]
+        [Trait(AcceptanceType, CheckIn)]
+        public void GetTests_FilesNull_ThrowNullException()
+        {
+            //arrange           
+            string mapFilePath =  GetMapFilePath();
+
+            try
+            {
+                TestSetGenerator.GetTests(null, mapFilePath);
+            }
+            catch (ArgumentNullException e)
+            {
+                // assert  
+                Assert.Contains("The files should never be null.", e.Message);
+                return;
+            }
+            throw new Exception("No exception was thrown.");
+        }
+
+        [Fact]
+        [Trait(AcceptanceType, CheckIn)]
+        public void GetTests_EmptyListOfFiles_ShouldReturnAllTests()
+        {
+            //arrange
+            HashSet<string> paths = new HashSet<string>() { };
+            string mapFilePath = GetMapFilePath();
+            int expectedNumberFiles = 51;
+            HashSet<string> actual;
+
+            //act
+            actual = (HashSet<string>)(TestSetGenerator.GetTests(paths, mapFilePath));
+
+            //assert            
+            Assert.True(expectedNumberFiles == actual.Count);
+        }
+
+        [Fact]
+        [Trait(AcceptanceType, CheckIn)]
+        public void GetTests_InvalidPath_ThrowNotNullException()
+        {
+            //arrange
+            HashSet<string> paths = new HashSet<string>()
+            {
+                "src /path1",
+                "src/path2",
+                "random3"
+            };
+            string mapFilePath = @"random";
+
```

Comment:
Please apply vertical spacing rules across this file.  There should only be an extra line after each closing brace ```}```, otherwise, please do not skip lines